tests/e2e/cmd/setup/upgrade/setup.go

```diff
@@ -12,11 +12,15 @@ import (
 	"github.com/titantkx/titan/testutil/cmd/keys"
 )
 
-const UpgradeName = "v2_0_1"
+const UpgradeName = "v3_0_0_rc_0"
 
 func Setup(m *testing.M, rootDir string, logger io.Writer) {
 	t := testutil.NewMockTest(os.Stderr)
 	defer t.Finish()
+	t.HandleOSInterrupt(func() {
+		defer setup.StopChain(t, logger, "docker-compose-genesis.yml")
+		defer setup.StopChain(t, logger, "docker-compose-local.yml")
+	})
```

Comment:
```suggestion

	// Set up signal handling for Ctrl+C
	sigChan := make(chan os.Signal, 1)
	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)

	go func() {
		fmt.Println("Received interrupt signal")
		setup.StopChain(t, logger, "docker-compose-genesis.yml")
		setup.StopChain(t, logger, "docker-compose-local.yml")
		os.Exit(1)
	}()
```

Reply:
Don't need to cleanup tests on interrupt signal

Reply:
I moved `HandleOSInterrupt` to `testutil/os`